test/testdata/namer/duplicate_field_error.rb

```diff
@@ -11,6 +11,7 @@ def initialize(x:, x: nil)
                    # ^^ error: Malformed `sig`
                    # ^^^^^^ error: duplicate argument name x
     @x = T.let(x, T.nilable(String))
+       # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Argument does not have asserted type
```

Comment:
Why does this change?

Reply:
This first `let` gets dropped during error recovery, while the second one is present for both parses. Here's the desugar tree before this PR:

```
class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
    <self>.extend(<emptyTree>::<C T>::<C Sig>)

    <self>.sig() do ||
      <self>.params(:x, <emptyTree>::<C Integer>, :x, <emptyTree>::<C String>).void()
    end

    def initialize<<todo method>>(x:, x$1: = nil, &<blk>)
      @x = <emptyTree>::<C T>.let(x, <emptyTree>::<C T>.nilable(<emptyTree>::<C String>))
    end
  end
end
```

and after:

```
class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
    <self>.extend(<emptyTree>::<C T>::<C Sig>)

    <self>.sig() do ||
      <self>.params(:x, <emptyTree>::<C Integer>, :x, <emptyTree>::<C String>).void()
    end

    def initialize<<todo method>>(x:, x$1: = nil, &<blk>)
      begin
        @x = <emptyTree>::<C T>.let(x, <emptyTree>::<C T>.nilable(<emptyTree>::<C String>))
        @x = <emptyTree>::<C T>.let(x, <emptyTree>::<C T>.nilable(<emptyTree>::<C String>))
      end
    end
  end
end
```